### ISSUE TITLE
regex library?

### ISSUE THREAD

**Issue by GasparVardanyan** [martanne/vis#831]
Why you don't move the regex code into a separate repository and make a structural regex library?

**Comment by martanne** [comment]
Do you have a clear use case in mind?

One of the main benefits of vis is that one can *interactively* refine selections by combining structural regular expression commands with more traditional modal editing. At the moment these are somewhat coupled together, but one could certainly envision a more rigid separation.

In terms of API it would at least take as input:

- structural regex command to apply
- text/string on which to operate
- set of selections (dots in the sam nomenclature)
- register values
- buffer list, if you want to support the `X` and `Y` commands
- ...

and produce updated versions of these or an error. The [existing API](https://github.com/martanne/vis/blob/686ba1c25831f09745c8492cfc38b852c19ef6b2/sam.h#L27) makes all these things (and probably more implicitly used ones) available through the `Vis` struct.

To summarize, while in principle a further refactoring would be possible, the immediate benefits aren't obvious for now.

**Comment by 3dc1d3** [comment]
@GasparVardanyan see also https://github.com/martanne/vis/issues/799
